pkg/utils/build/source_image.go

```diff
@@ -165,3 +176,274 @@ func IsPreFetchDependencysFilesExists(absExtraSourceDirPath string, isHermetic b
+		if resp, err := http.Get(dockerfile); err == nil {
+			defer resp.Body.Close()
+			if body, err := io.ReadAll(resp.Body); err == nil {
+				return body, err
+			} else {
+				return nil, err
+			}
+		} else {
+			return nil, err
+		}
+	}
+	return nil, fmt.Errorf(
+		fmt.Sprintf("resolveDockerfile: can't resolve Dockerfile from path context %s and dockerfile %s",
+			pathContext, dockerfile),
+	)
+}
+
+// ReadDockerfileUsedForBuild reads the Dockerfile and return its content.
+func ReadDockerfileUsedForBuild(c client.Client, tektonController *tekton.TektonController, pr *pipeline.PipelineRun) ([]byte, error) {
+	var paramDockerfileValue, paramPathContextValue string
+	var paramUrlValue, paramRevisionValue string
+	var err error
+	getParam := tektonController.GetTaskRunParam
+
+	if paramDockerfileValue, err = getParam(c, pr, "build-container", "DOCKERFILE"); err != nil {
+		return nil, err
+	}
+
+	if paramPathContextValue, err = getParam(c, pr, "build-container", "CONTEXT"); err != nil {
+		return nil, err
+	}
+
+	// get git-clone param url and revision
+	if paramUrlValue, err = getParam(c, pr, "clone-repository", "url"); err != nil {
+		return nil, err
+	}
+
+	if paramRevisionValue, err = getParam(c, pr, "clone-repository", "revision"); err != nil {
+		return nil, err
+	}
+
+	dockerfileContent, err := readDockerfile(paramPathContextValue, paramDockerfileValue, paramUrlValue, paramRevisionValue)
+	if err != nil {
+		return nil, err
+	}
+	return dockerfileContent, nil
+}
+
+type SourceBuildResult struct {
+	Status                  string `json:"status"`
+	Message                 string `json:"message,omitempty"`
+	DependenciesIncluded    bool   `json:"dependencies_included"`
+	BaseImageSourceIncluded bool   `json:"base_image_source_included"`
+	ImageUrl                string `json:"image_url"`
+	ImageDigest             string `json:"image_digest"`
+}
+
+// ReadSourceBuildResult reads source-build task result BUILD_RESULT and returns the decoded data.
+func ReadSourceBuildResult(c client.Client, tektonController *tekton.TektonController, pr *pipeline.PipelineRun) (*SourceBuildResult, error) {
+	sourceBuildResult, err := tektonController.GetTaskRunResult(c, pr, "build-source-image", "BUILD_RESULT")
+	if err != nil {
+		return nil, err
+	}
+	var buildResult SourceBuildResult
+	if err = json.Unmarshal([]byte(sourceBuildResult), &buildResult); err != nil {
+		return nil, err
+	}
+	return &buildResult, nil
+}
+
+type Dockerfile struct {
+	parsedContent *parser.Result
+}
+
+func ParseDockerfile(content []byte) (*Dockerfile, error) {
+	parsedContent, err := parser.Parse(bytes.NewReader(content))
+	if err != nil {
+		return nil, err
+	}
+	df := Dockerfile{
+		parsedContent: parsedContent,
+	}
+	return &df, nil
+}
+
+func (d *Dockerfile) ParentImages() []string {
+	parentImages := make([]string, 0, 5)
+	for _, child := range d.parsedContent.AST.Children {
+		if child.Value == "FROM" {
+			parentImages = append(parentImages, child.Next.Value)
```

Comment:
Does this work correctly for `FROM ... AS ...`?

Reply:
Yes